packages/database/src/migrations/20210202061216-AddFetpGraduateIndividualProfiles-modifies-data.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+import { arrayToDbString, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const dataServices = [{ isDataRegional: true }];
+
+const buildTable = report => {
+  const mapElementsToCells = element =>
+    report.elementConfigs && element in report.elementConfigs
+      ? [report.elementConfigs[element]]
+      : [element];
+
+  const dataBuilderConfig = {
+    rows: Object.values(report.elements),
+    cells: Object.keys(report.elements).map(mapElementsToCells),
+    columns: ['main'],
+    entityAggregation: {
+      aggregationType: 'RAW',
+      dataSourceEntityType: ['individual'],
+      aggregationEntityType: 'individual',
+    },
+    noDataValue: '',
+  };
+
+  const viewJson = {
+    name: report.name,
+    type: 'view',
+    viewType: 'multiValue',
+    valueType: 'text',
+  };
+
+  return {
+    id: report.id,
+    dataBuilder: report.dataBuilder,
+    dataBuilderConfig,
+    viewJson,
+    dataServices,
+  };
+};
+
+const buildSingle = report => {
+  const dataBuilderConfig = {
+    dataElementCodes: report.elements,
+  };
+  const viewJson = {
+    name: report.name,
+    type: 'view',
+    viewType: 'singleValue',
+    noDataMessage: ' ',
```

Comment:
Does this need any `web-frontend` updates, or will it work out of the box? Just asking because you mentioned

> Add empty noDataMessage for SingleValue and MultiSingleValue reports

in the PR's description, so I assumed this would be related to `web-frontend` changes.

Reply:
Yes using existing functionality, I'll remove from description as it is confusing, I copied from commit messages.

Reply:
One 'hack' I had to  do was to set empty string with space (' ') because  '' was being falsey, but feel this was an ok solution.

Reply:
Hm ideally the front-end code should be updated to check for `null`/`undefined` and not for all falsy values? Do you thing that would be a tricky change?

Reply:
No should be fine as long as it doesn't break existing frontend component config assumptions. (because it is changed in code rather than just a config setting) But I can check this.
```
export const NoDataMessage = ({ viewContent }) => {
  let message = 'No data';

  if (viewContent.noDataMessage) {
    message = viewContent.noDataMessage;
  } else if (viewContent.source === 'mSupply') {
    message = 'Requires mSupply';
  } else ...
  }
```

Reply:
Yeah, `''` and `0` should both be valid no data messages

Reply:
Changed to `if ('noDataMessage' in viewContent)`